src/govuk/components/pagination/template.test.js

```diff
@@ -0,0 +1,158 @@
+    })
+
+    it('renders the correct URLs for each link', () => {
+      const $ = render('pagination', examples.default)
+      const previous = $('.govuk-pagination__prev .govuk-pagination__link')
+      const next = $('.govuk-pagination__next .govuk-pagination__link')
+      const firstNumber = $('.govuk-pagination__item:first-child .govuk-pagination__link')
+      const secondNumber = $('.govuk-pagination__item:nth-child(2) .govuk-pagination__link')
+      const thirdNumber = $('.govuk-pagination__item:last-child .govuk-pagination__link')
+
+      expect(previous.attr('href')).toEqual('/previous')
+      expect(next.attr('href')).toEqual('/next')
+      expect(firstNumber.attr('href')).toEqual('/page/1')
+      expect(secondNumber.attr('href')).toEqual('')
+      expect(thirdNumber.attr('href')).toEqual('/page/3')
+    })
+
+    it('renders the correct number within each pagination item', () => {
+      const $ = render('pagination', examples.default)
+      const firstNumber = $('.govuk-pagination__item:first-child')
+      const secondNumber = $('.govuk-pagination__item:nth-child(2)')
+      const thirdNumber = $('.govuk-pagination__item:last-child')
+
+      expect(firstNumber.text().trim()).toEqual('1')
+      expect(secondNumber.text().trim()).toEqual('2')
+      expect(thirdNumber.text().trim()).toEqual('3')
+    })
+
+    // The current item is marked up with a visually hidden span and an aria-hidden span side by side
+    // Instead of the aria-label solution used for the links in the pagination because of issues caused
+    // by aria-label on non-interactive elements like li's
+    it('marks up the current item correctly', () => {
+      const $ = render('pagination', examples.default)
+      const currentNumber = $('.govuk-pagination__item--current')
+      const currentNumberLink = currentNumber.find('.govuk-pagination__link')
+
+      expect(currentNumber).toBeTruthy()
+      expect(currentNumberLink.attr('aria-current')).toEqual('page')
+    })
+
+    it('marks up the previous and next buttons correctly', () => {
+      const $ = render('pagination', examples.default)
+      const previousLink = $('.govuk-pagination__prev .govuk-pagination__link')
+      const nextLink = $('.govuk-pagination__next .govuk-pagination__link')
+      const previousSvg = $('.govuk-pagination__icon--prev')
+      const nextSvg = $('.govuk-pagination__icon--next')
+
+      expect(previousLink.attr('rel')).toEqual('prev')
+      expect(previousSvg.attr('height')).toEqual('13')
+      expect(previousSvg.attr('width')).toEqual('15')
+      expect(previousSvg.attr('aria-hidden')).toEqual('true')
+      expect(previousSvg.attr('focusable')).toEqual('false')
+      expect(nextLink.attr('rel')).toEqual('next')
+      expect(nextSvg.attr('height')).toEqual('13')
+      expect(nextSvg.attr('width')).toEqual('15')
+      expect(nextSvg.attr('aria-hidden')).toEqual('true')
+      expect(nextSvg.attr('focusable')).toEqual('false')
+    })
```

Comment:
I'd suggest mimicking the tests we have in the header for the focusable and aria-hidden attributes and do something like this:

```
    it('sets focusable="false" so that IE does not treat it as an interactive element', () => {
      expect($svg.attr('focusable')).toEqual('false')
    })

    it('sets aria-hidden="true" so that it is ignored by assistive technologies', () => {
      expect($svg.attr('aria-hidden')).toEqual('true')
    })
```

I really like that we're testing the rel attributes, but not sure what we gain from testing width and height TBH.

Reply:
I've re-worked it so that there are some more `describe`'s and it's segmented into more specific testing suites. LEt me know what you think.